src/litdata/streaming/client.py

```diff
@@ -37,7 +37,7 @@ def _create_client(self) -> None:
             os.getenv("AWS_SHARED_CREDENTIALS_FILE") == os.getenv("AWS_CONFIG_FILE") == "/.credentials/.aws_credentials"
         )
 
-        if has_shared_credentials_file or not _IS_IN_STUDIO:
+        if has_shared_credentials_file or not _IS_IN_STUDIO or self._storage_options:
```

Comment:
It might be better to propagate the storage options to the client below. I found the provider one to be more reliable in general.